diff --git a/tests/unit/test_convert.py b/tests/unit/test_convert.py
index 9b42431ef896b8e748ade0eac03511cc0ea15ac5..d6e072db40b7997b394e864acc9d45a0df23eb9a 100755
--- a/tests/unit/test_convert.py
+++ b/tests/unit/test_convert.py
@@ -12,7 +12,7 @@ g_max_iso = "-".join([f"{i:02}" for i in g_max])
 
 
 def test_importing_at_init_module():
-    from hijridate import Gregorian, Hijri
+    from hijridate import Gregorian, Hijri  # noqa: PLC0415
 
     assert Hijri(1410, 8, 13)
     assert Gregorian(1990, 3, 10)
@@ -87,6 +87,7 @@ class TestHijri:
         assert self.hijri_date.month_name() == "Sha'ban"
         assert self.hijri_date.month_name("en") == "Sha'ban"
         assert self.hijri_date.month_name("en-US") == "Sha'ban"
+        assert self.hijri_date.month_name("tr") == "Şaban"
 
     def test_weekday(self):
         assert self.hijri_date.weekday() == 5
@@ -98,11 +99,13 @@ class TestHijri:
         assert self.hijri_date.day_name() == "Saturday"
         assert self.hijri_date.day_name("en") == "Saturday"
         assert self.hijri_date.day_name("en-US") == "Saturday"
+        assert self.hijri_date.day_name("tr") == "Cumartesi"
 
     def test_notation(self):
         assert self.hijri_date.notation() == "AH"
         assert self.hijri_date.notation("en") == "AH"
         assert self.hijri_date.notation("en-US") == "AH"
+        assert self.hijri_date.notation("tr") == "Hicri"
 
     def test_to_julian(self):
         assert self.hijri_date.to_julian() == 2447961
@@ -180,16 +183,19 @@ class TestGregorian:
         assert self.gregorian_date.month_name() == "March"
         assert self.gregorian_date.month_name("en") == "March"
         assert self.gregorian_date.month_name("en-US") == "March"
+        assert self.gregorian_date.month_name("tr") == "Mart"
 
     def test_day_name(self):
         assert self.gregorian_date.day_name() == "Saturday"
         assert self.gregorian_date.day_name("en") == "Saturday"
         assert self.gregorian_date.day_name("en-US") == "Saturday"
+        assert self.gregorian_date.day_name("tr") == "Cumartesi"
 
     def test_notation(self):
         assert self.gregorian_date.notation() == "CE"
         assert self.gregorian_date.notation("en") == "CE"
         assert self.gregorian_date.notation("en-US") == "CE"
+        assert self.gregorian_date.notation("tr") == "Miladi"
 
     def test_to_julian(self):
         assert self.gregorian_date.to_julian() == 2447961
