diff --git a/tests/test_client.py b/tests/test_client.py
index 6cdb5b5c4e621be1743d9cd4b8d373f8be2d9393..82f90dc5ab7c53a5ec51fb3874c03059b3def0a5 100644
--- a/tests/test_client.py
+++ b/tests/test_client.py
@@ -20,7 +20,7 @@ import uuid
 import zlib
 
 import pytest
-from urllib3 import HTTPConnectionPool
+from urllib3 import HTTPConnectionPool, HTTPResponse as URLLib3HTTPResponse
 
 from newrelic_telemetry_sdk.client import (
     EventClient,
@@ -89,13 +89,15 @@ def capture_request(fn):
 
 
 def disable_sending(*args, **kwargs):
-    response = HTTPResponse(status=202)
-    response.request = Request(*args, **kwargs)
+    urllib3_response = URLLib3HTTPResponse(status=202)
+    urllib3_response.request = Request(*args, **kwargs)
+    response = HTTPResponse(urllib3_response)
     return response
 
 
 def test_response_json():
-    response = HTTPResponse(status=200, body=b"{}")
+    urllib3_response = URLLib3HTTPResponse(status=200, body=b"{}")
+    response = HTTPResponse(urllib3_response)
     assert response.json() == {}
 
 
@@ -103,21 +105,30 @@ def test_response_json():
     "status,expected", ((199, False), (200, True), (299, True), (300, False))
 )
 def test_response_ok(status, expected):
-    response = HTTPResponse(status=status)
+    urllib3_response = URLLib3HTTPResponse(status=status)
+    response = HTTPResponse(urllib3_response)
     assert response.ok is expected
 
 
 def test_response_raise_for_status_error():
-    response = HTTPResponse(status=500)
+    urllib3_response = URLLib3HTTPResponse(status=500)
+    response = HTTPResponse(urllib3_response)
     with pytest.raises(HTTPError):
         response.raise_for_status()
 
 
 def test_response_raise_for_status_ok():
-    response = HTTPResponse(status=200)
+    urllib3_response = URLLib3HTTPResponse(status=200)
+    response = HTTPResponse(urllib3_response)
     response.raise_for_status()
 
 
+def test_wrapped_response_attributes_available():
+    urllib3_response = URLLib3HTTPResponse(status=200)
+    response = HTTPResponse(urllib3_response)
+    assert response.status == 200
+
+
 @pytest.fixture
 def span_client(request, monkeypatch):
     host = os.environ.get("NEW_RELIC_HOST", "")
diff --git a/tests/test_client_proxy.py b/tests/test_client_proxy.py
index 2842c9e61654019a574d6bfc94ddf73c268fa640..081400ba99a5f12b78729b6f9dfbfe1beca3936c 100644
--- a/tests/test_client_proxy.py
+++ b/tests/test_client_proxy.py
@@ -2,7 +2,7 @@ import socket
 import threading
 import functools
 import logging
-from urllib3 import HTTPConnectionPool, exceptions
+from urllib3 import HTTPConnectionPool, exceptions, HTTPResponse
 
 try:
     from http.server import BaseHTTPRequestHandler, HTTPServer
@@ -14,7 +14,6 @@ from newrelic_telemetry_sdk.client import (
     SpanClient,
     MetricClient,
     EventClient,
-    HTTPResponse,
 )
 import pytest
 
@@ -78,12 +77,17 @@ def force_response(fn):
     def wrapper(*args, **kwargs):
         # Make the actual call to urlopen so urllib3 will attempt a connect.
         # Since the proxy does not actually open a socket to the remote host we
-        # expect cPython to raise a ProxyError and PyPy will raise an OSError or
-        # SSLError. For testing these errors can be ignored, we just care that the
-        # connect occured with the proper request line.
+        # expect cPython to raise a ProxyError or ProtocolError and PyPy will
+        # raise an OSError or SSLError. For testing these errors can be ignored,
+        # we just care that the connect occured with the proper request line.
         try:
             fn(*args, **kwargs)
-        except (exceptions.ProxyError, exceptions.SSLError, OSError):
+        except (
+            exceptions.ProxyError,
+            exceptions.ProtocolError,
+            exceptions.SSLError,
+            OSError,
+        ):
             pass
 
         response = HTTPResponse(status=202)
