diff --git a/tests/core/test_flow.py b/tests/core/test_flow.py
index a4a9bdf2473dfd394b63348e12fc0546fdba9956..3238cf05ec23ae4432ae1c9dadbd166482afc9f6 100644
--- a/tests/core/test_flow.py
+++ b/tests/core/test_flow.py
@@ -1427,6 +1427,17 @@ def test_job_autoreplace_dict(memory_jobstore):
     assert 5 in all_outputs  # sum
     assert 1 in all_outputs  # diff
 
+    # check that if an empty dict is returned the output is not misinterpreted
+    # as a replace
+    @job
+    def make_empty_dict():
+        return {}
+
+    job1 = make_empty_dict()
+    flow = Flow([job1])
+    responses = run_locally(flow, store=memory_jobstore, ensure_success=True)
+    assert responses[job1.uuid][1].output == {}
+
 
 def test_get_item():
     from jobflow import Flow, job, run_locally
diff --git a/tests/core/test_job.py b/tests/core/test_job.py
index d1c8f7f2d970b44fc0d92519fbd557d13a4ea066..727d750d672c02853eb3800ad050189269c2bc73 100644
--- a/tests/core/test_job.py
+++ b/tests/core/test_job.py
@@ -135,6 +135,7 @@ def test_job_run(capsys, memory_jobstore, memory_data_jobstore):
 
 def test_replace_response(memory_jobstore):
     from jobflow import Flow, Job, Response
+    from jobflow.core.job import store_inputs
 
     def replace_job():
         job = Job(add, function_args=(1,))
@@ -183,9 +184,11 @@ def test_replace_response(memory_jobstore):
     test_job = Job(replace_list_job, metadata=metadata, output_schema="123")
     response = test_job.run(memory_jobstore)
     assert isinstance(response.replace, Flow)
-    assert response.replace[-1].function == add
-    assert len(response.replace) == 2
-    # currently output schema and metadata ignored in this case
+    assert response.replace[-2].function == add
+    assert response.replace[-1].function == store_inputs.original
+
+    assert len(response.replace) == 3
+    # currently output schema and metadata ignored for all but the last `store_inputs`
     for j in response.replace:
         assert j.hosts == [response.replace.uuid]
 
@@ -218,12 +221,13 @@ def test_replace_response(memory_jobstore):
     test_job = Job(replace_list_flow, metadata=metadata, output_schema="123")
     response = test_job.run(memory_jobstore)
     assert isinstance(response.replace, Flow)
-    assert isinstance(response.replace[-1], Flow)
-    assert len(response.replace) == 2
+    assert isinstance(response.replace[-2], Flow)
+    assert len(response.replace) == 3
     for f in response.replace:
-        for j in f:
-            assert j.hosts == [f.uuid, response.replace.uuid]
-    # currently output schema and metadata ignored in this case
+        if isinstance(f, Flow):
+            for j in f:
+                assert j.hosts == [f.uuid, response.replace.uuid]
+    # currently output schema and metadata ignored for all but the last `store_inputs`
 
 
 def test_job_config(memory_jobstore):
@@ -352,8 +356,9 @@ def test_job_config(memory_jobstore):
     # test replace with flow
     test_job = Job(replace_flow, config=nopass_config)
     response = test_job.run(memory_jobstore)
-    for j in response.replace:
+    for j in response.replace[:-1]:
         assert j.config.manager_config == {}
+    assert response.replace[-1].config.manager_config == manager_config2
 
     test_job = Job(replace_flow, config=pass_config)
     response = test_job.run(memory_jobstore)
