diff --git a/index.test.js b/index.test.js
index c619935789c95103ef0a9fa147f3dd1bce816665..afa96944a6b579136f239f29523bf1f85868d0a7 100644
--- a/index.test.js
+++ b/index.test.js
@@ -618,21 +618,20 @@ div {
     '/* Comment with ^ $ . | ? * + () */ div[data-roots-all^=1] * #id .class {}')
 })
 
-// !
-// test("Save the parent's comment with newline", () => {
-//   run(
-//     `
-// a { 
-//   /*i*/
-
-//   /*i2*/
-//   b {} 
-//   /*i3*/
-//   s {} 
-// }`,
-//     `a { /*i*/ } /*i2*/ a b {} /*i3*/ a s {}`
-//   )
-// })
+test("Save several rules with attached comments", () => {
+  run(
+    `
+a { 
+  /*i*/
+
+  /*i2*/
+  b {} 
+  /*i3*/
+  s {} 
+}`,
+    `a { /*i*/ } /*i2*/ a b {} /*i3*/ a s {}`
+  )
+})
 
 test("Save the parent's comment with newline", () => {
   run(
@@ -647,10 +646,12 @@ test("Save the parent's comment with newline", () => {
 
 test('Save the comments for the parent and child', () => {
   run(
-    `a { 
-    /*i*/ 
-    /*o*/
-    b {} }`,
+    `
+a { 
+  /*i*/ 
+  /*o*/
+  b {} 
+}`,
 
     `a { /*i*/ } /*o*/ a b {}`
   )
