diff --git a/tests/test_pyphen.py b/tests/test_pyphen.py
index ab1db991b075207d11bef3bd981ce4c29dc48f22..9ad5a018f531c37ff3524411e4e93ae202e85844 100644
--- a/tests/test_pyphen.py
+++ b/tests/test_pyphen.py
@@ -8,6 +8,8 @@ Tests can be launched with Pytest.
 """
 
 
+from pathlib import Path
+
 import pyphen
 
 
@@ -57,6 +59,20 @@ def test_personal_dict():
     assert dic.inserted('autobandventieldopje') == 'au-to-band-ven-tiel-dop-je'
 
 
+def test_dict_from_filename():
+    """Test a dict open from filename."""
+    dic_path = Path(__file__).parents[1] / 'pyphen' / 'dictionaries' / 'hyph_fr.dic'
+    dic = pyphen.Pyphen(filename=str(dic_path))
+    assert dic.inserted('bonjour') == 'bon-jour'
+
+
+def test_dict_from_path():
+    """Test a dict open from path."""
+    dic_path = Path(__file__).parents[1] / 'pyphen' / 'dictionaries' / 'hyph_fr.dic'
+    dic = pyphen.Pyphen(filename=dic_path)
+    assert dic.inserted('bonjour') == 'bon-jour'
+
+
 def test_left_right():
     """Test the ``left`` and ``right`` parameters."""
     dic = pyphen.Pyphen(lang='nl_NL')
