diff --git a/src/testfixtures/shouldraise.py b/src/testfixtures/shouldraise.py
index 64c548f..4ecdd3a 100644
--- a/src/testfixtures/shouldraise.py
+++ b/src/testfixtures/shouldraise.py
@@ -6,6 +6,7 @@ from typing import Callable, TypeAlias, Iterator, Self, ParamSpec, TypeVar, Gene
 
 from testfixtures import diff, compare, not_there, singleton
 from .comparers import split_repr
+from .comparison import RenderingComparison
 
 
 param_docs = """
@@ -111,6 +112,9 @@ class ShouldRaise(Generic[E]):
                         if self.exception is not actual_:
                             return False
                         self._check_match(actual)
+                    elif isinstance(self.exception, RenderingComparison):
+                        if self.exception.type is not type(actual):
+                            return False
                     else:
                         if type(self.exception) is not type(actual):
                             return False
