diff --git a/test/query.test.ts b/test/query.test.ts
index 4fbe0c426ba25c10f2d65f9fc5f4706d0343b65c..a9c2e7b9e7a6d96433702cefb170f33f18ed84b7 100644
--- a/test/query.test.ts
+++ b/test/query.test.ts
@@ -86,6 +86,7 @@ describe("getQuery", () => {
     "http://foo.com/?param=%7B%22a%22:%5B%7B%22obj%22:%5B1,2,3%5D%7D%5D%7D": {
       param: '{"a":[{"obj":[1,2,3]}]}',
     },
+    "http://foo.com/?toString=foo": { toString: "foo" },
   };
 
   for (const t in tests) {
