diff --git a/test/defaults.test.ts b/test/defaults.test.ts
index d1a21175a78a71f016858514caf0ce1f6533d6d1..9a9381cd1de3ed46eaef2ac5a4235f94b767edc7 100644
--- a/test/defaults.test.ts
+++ b/test/defaults.test.ts
@@ -1,5 +1,5 @@
 import { describe, expect, it } from "vitest";
-import fetchMock from "fetch-mock";
+import fetchMock, { type CallLog } from "fetch-mock";
 import { getUserAgent } from "universal-user-agent";
 
 import { VERSION } from "../src/version";
@@ -214,4 +214,118 @@ describe("graphql.defaults()", () => {
       },
     });
   });
+
+  it("Allows user to specify non variable options", () => {
+    const mockData = {
+      repository: {
+        issues: {
+          edges: [
+            {
+              node: {
+                title: "Foo",
+              },
+            },
+            {
+              node: {
+                title: "Bar",
+              },
+            },
+            {
+              node: {
+                title: "Baz",
+              },
+            },
+          ],
+        },
+      },
+    };
+
+    const query = /* GraphQL */ `
+        query Blue($last: Int) {
+          repository(owner: "blue", name: "graphql.js") {
+            issues(last: $last) {
+              edges {
+                node {
+                  title
+                }
+              }
+            }
+          }
+        }
+
+        query Green($last: Int) {
+          repository(owner: "green", name: "graphql.js") {
+            issues(last: $last) {
+              edges {
+                node {
+                  title
+                }
+              }
+            }
+          }
+        }
+        `.trim();
+
+    const fetch = fetchMock.createInstance();
+
+    fetch.post(
+      "https://api.github.com/graphql",
+      { data: mockData },
+      {
+        method: "POST",
+        headers: {
+          accept: "application/vnd.github.v3+json",
+          authorization: "token secret123",
+          "user-agent": userAgent,
+        },
+        matcherFunction: (callLog: CallLog) => {
+          const expected = {
+            query: query,
+            operationName: "Blue",
+            variables: { last: 3 },
+          };
+          const result = callLog.options.body === JSON.stringify(expected);
+          if (!result) {
+            console.warn("Body did not match expected value", {
+              expected,
+              actual: JSON.parse(callLog.options.body as string),
+            });
+          }
+          return result;
+        },
+      },
+    );
+
+    const authenticatedGraphql = graphql.defaults({
+      headers: {
+        authorization: `token secret123`,
+      },
+      request: {
+        fetch: fetch.fetchHandler,
+      },
+    });
+
+    return new Promise<void>((res, rej) =>
+      authenticatedGraphql({
+        query,
+        headers: {
+          authorization: `token secret123`,
+        },
+        request: {
+          fetch: fetch.fetchHandler,
+        },
+        operationName: "Blue",
+        last: 3,
+      })
+        .then((result) => {
+          expect(JSON.stringify(result)).toStrictEqual(
+            JSON.stringify(mockData),
+          );
+          res();
+        })
+        .catch(() => {
+          rej("Should have resolved");
+        }),
+    );
+  });
 });
diff --git a/test/graphql.test.ts b/test/graphql.test.ts
index 435247f69e99e80fe0b02db5ffa93b8615800a66..4c8269ba0afee2f96ca1af907697ed2eb6e01471 100644
--- a/test/graphql.test.ts
+++ b/test/graphql.test.ts
@@ -1,5 +1,5 @@
 import { describe, expect, it } from "vitest";
-import fetchMock from "fetch-mock";
+import fetchMock, { type CallLog } from "fetch-mock";
 import { getUserAgent } from "universal-user-agent";
 
 import { graphql } from "../src";
@@ -323,4 +323,164 @@ describe("graphql()", () => {
       );
     });
   });
+
+  describe("When using a query with multiple operations", () => {
+    const mockData = {
+      repository: {
+        issues: {
+          edges: [
+            {
+              node: {
+                title: "Foo",
+              },
+            },
+            {
+              node: {
+                title: "Bar",
+              },
+            },
+            {
+              node: {
+                title: "Baz",
+              },
+            },
+          ],
+        },
+      },
+    };
+
+    const mockErrors = {
+      errors: [{ message: "An operation name is required" }],
+      data: undefined,
+      status: 400,
+    };
+
+    const query = /* GraphQL */ `
+      query Blue($last: Int) {
+        repository(owner: "blue", name: "graphql.js") {
+          issues(last: $last) {
+            edges {
+              node {
+                title
+              }
+            }
+          }
+        }
+      }
+
+      query Green($last: Int) {
+        repository(owner: "green", name: "graphql.js") {
+          issues(last: $last) {
+            edges {
+              node {
+                title
+              }
+            }
+          }
+        }
+      }
+      `.trim();
+
+    it("Sends both queries to the server (which will respond with bad request)", () => {
+      const fetch = fetchMock.createInstance();
+
+      fetch.post("https://api.github.com/graphql", mockErrors, {
+        method: "POST",
+        headers: {
+          accept: "application/vnd.github.v3+json",
+          authorization: "token secret123",
+          "user-agent": userAgent,
+        },
+        matcherFunction: (callLog: CallLog) => {
+          const expected = {
+            query: query,
+            variables: { last: 3 },
+          };
+          const result = callLog.options.body === JSON.stringify(expected);
+          if (!result) {
+            console.warn("Body did not match expected value", {
+              expected,
+              actual: JSON.parse(callLog.options.body as string),
+            });
+          }
+          return result;
+        },
+      });
+
+      return new Promise<void>((res, rej) =>
+        graphql(query, {
+          headers: {
+            authorization: `token secret123`,
+          },
+          request: {
+            fetch: fetch.fetchHandler,
+          },
+          last: 3,
+        })
+          .then(() => {
+            rej("Should have thrown an error");
+          })
+          .catch((result) => {
+            expect(JSON.stringify(result.response)).toStrictEqual(
+              JSON.stringify(mockErrors),
+            );
+            res();
+          }),
+      );
+    });
+
+    it('Allows the user to specify the operation name in the "operationName" option', () => {
+      const fetch = fetchMock.createInstance();
+
+      fetch.post(
+        "https://api.github.com/graphql",
+        { data: mockData },
+        {
+          method: "POST",
+          headers: {
+            accept: "application/vnd.github.v3+json",
+            authorization: "token secret123",
+            "user-agent": userAgent,
+          },
+          matcherFunction: (callLog: CallLog) => {
+            const expected = {
+              query: query,
+              operationName: "Blue",
+              variables: { last: 3 },
+            };
+            const result = callLog.options.body === JSON.stringify(expected);
+            if (!result) {
+              console.warn("Body did not match expected value", {
+                expected,
+                actual: JSON.parse(callLog.options.body as string),
+              });
+            }
+            return result;
+          },
+        },
+      );
+
+      return new Promise<void>((res, rej) =>
+        graphql(query, {
+          headers: {
+            authorization: `token secret123`,
+          },
+          request: {
+            fetch: fetch.fetchHandler,
+          },
+          operationName: "Blue",
+          last: 3,
+        })
+          .then((result) => {
+            expect(JSON.stringify(result)).toStrictEqual(
+              JSON.stringify(mockData),
+            );
+            res();
+          })
+          .catch((error) => {
+            rej(error);
+          }),
+      );
+    });
+  });
 });
