diff --git a/bottle.py b/bottle.py
index 053f9b4..9ef3bba 100755
--- a/bottle.py
+++ b/bottle.py
@@ -2820,17 +2820,18 @@ def static_file(filename, root,
                                    clen, filename)
         etag = hashlib.sha1(tob(etag)).hexdigest()
 
+    inm = getenv('HTTP_IF_NONE_MATCH')
     if etag:
         headers['ETag'] = etag
-        check = getenv('HTTP_IF_NONE_MATCH')
-        if check and check == etag:
+        if inm and inm == etag:
             return HTTPResponse(status=304, **headers)
 
-    ims = getenv('HTTP_IF_MODIFIED_SINCE')
-    if ims:
-        ims = parse_date(ims.split(";")[0].strip())
-        if ims is not None and ims >= int(stats.st_mtime):
-            return HTTPResponse(status=304, **headers)
+    if not inm:
+        ims = getenv('HTTP_IF_MODIFIED_SINCE')
+        if ims:
+            ims = parse_date(ims.split(";")[0].strip())
+            if ims is not None and ims >= int(stats.st_mtime):
+                return HTTPResponse(status=304, **headers)
 
     body = '' if request.method == 'HEAD' else open(filename, 'rb')
 
