diff --git a/test/test_bibliography.py b/test/test_bibliography.py
index b2f9ee90d87a021e9a7c1a16af93ad47f72b144e..a5a3f6c7687699201cdf6467b82f1f7d5c0b621f 100644
--- a/test/test_bibliography.py
+++ b/test/test_bibliography.py
@@ -43,13 +43,12 @@ def test_bibliography_header(app, warning) -> None:
         "bibtex_footbibliography_header": ".. rubric:: Footnote Citations",
     },
 )
-def test_bibliography_empty_no_header(app, warning) -> None:
+def test_bibliography_empty_has_header(app, warning) -> None:
     app.build()
     assert not warning.getvalue()
     output = (app.outdir / "index.pseudoxml").read_text(encoding="utf-8")
-    assert "Regular Citations" not in output
+    assert "Regular Citations" in output
     assert "Footnote Citations" not in output
-    assert "<rubric" not in output
 
 
 @pytest.mark.sphinx("html", testroot="bibliography_style_default")
