diff --git a/tests/main/test_main.py b/tests/main/test_main.py
index b1afb05a27df5e14d347c3f6be71dcddbe1221a2..8bc9f25e73745c2116131d2a4671e81e7c9f9fba 100644
--- a/tests/main/test_main.py
+++ b/tests/main/test_main.py
@@ -963,6 +963,17 @@ def test_main_single_file_client(
             "example_client",
             MODELS_ONLY_PATH / "example" / "expected_client",
         ),
+        (
+            (
+                MODELS_ONLY_PATH / "only_used_inputs_and_enums" / "pyproject.toml",
+                (
+                    MODELS_ONLY_PATH / "only_used_inputs_and_enums" / "schema.graphql",
+                    MODELS_ONLY_PATH / "only_used_inputs_and_enums" / "queries.graphql",
+                ),
+            ),
+            "example_client",
+            MODELS_ONLY_PATH / "only_used_inputs_and_enums" / "expected_client",
+        ),
     ],
     indirect=["project_dir"],
 )
