diff --git a/src/context.test.ts b/src/context.test.ts
index a8a5dd2ed044bdf5977a281d4f1f9b9c0910aa61..670a08a3ae3f18dd97a0eb7192cedca21fb3b960 100644
--- a/src/context.test.ts
+++ b/src/context.test.ts
@@ -144,6 +144,13 @@ describe('Context', () => {
     expect(foo).toBe('Bar, Buzz')
   })
 
+  it('c.set() and c.get() with symbol key', () => {
+    const sym = Symbol('key')
+    expect(c.get(sym)).toBe(undefined)
+    c.set(sym, 'value')
+    expect(c.get(sym)).toBe('value')
+  })
+
   it('c.set() and c.get()', async () => {
     expect(c.get('foo')).toBe(undefined)
     c.set('foo', 'bar')
