diff --git a/prefixdate/parse.py b/prefixdate/parse.py
index 1f89526..20ee3eb 100644
--- a/prefixdate/parse.py
+++ b/prefixdate/parse.py
@@ -11,7 +11,7 @@ log = logging.getLogger(__name__)
 Raw = Union[None, str, date, datetime, int, "DatePrefix"]
 
 REGEX = re.compile(
-    r"^\s*((?P<year>[12]\d{3})"
+    r"^\s*((?P<year>[12]\d{3})(?!\d)"
     r"(-(?P<month>\d{1,2})"
     r"(-(?P<day>\d{1,2})"
     r"([T ]"
