diff --git a/tests/plugins/ip_public_test.py b/tests/plugins/ip_public_test.py
index 5480f093b8a583f75fb798c31add82f748f4ff6d..77c9da95b7915ca54014382ef3ee658736c2bdf4 100644
--- a/tests/plugins/ip_public_test.py
+++ b/tests/plugins/ip_public_test.py
@@ -37,6 +37,7 @@ class TestIPPublicDetector:
                 ('10.0.0.1', False),
                 ('172.16.0.1', False),
                 ('192.168.0.1', False),
+                ('169.254.169.254', False),
                 # Invalid IPv4 addresses
                 ('256.256.256.256', False),
                 ('1.2.3', False),
