diff --git a/tests/test_asteval.py b/tests/test_asteval.py
index 2234c72490a5951b666644d0153d2e159c7fa06b..afe3ac047fd38f0d700e4c278b34d64508024257 100644
--- a/tests/test_asteval.py
+++ b/tests/test_asteval.py
@@ -12,6 +12,7 @@ from functools import partial
 from io import StringIO
 from sys import version_info
 from tempfile import NamedTemporaryFile
+from types import SimpleNamespace
 
 import pytest
 
@@ -590,6 +591,20 @@ def test_assignment(nested):
         interp('a[1:5] = 1 + 0.5 * arange(4)')
         isnear(interp, "a", np.array([0., 1., 1.5, 2., 2.5, 5., 6., 7., 8., 9.]))
 
+
+@pytest.mark.parametrize("nested", [False, True])
+@pytest.mark.parametrize("attr", ["__private__", "func_globals"])
+def test_unsafe_attribute_assignment(nested, attr):
+    """unsafe attribute assignment"""
+    obj = SimpleNamespace()
+    sym_table = make_symbol_table(nested=nested, obj=obj)
+    interp = Interpreter(symtable=sym_table)
+    interp(f"obj.{attr} = 1", show_errors=False)
+    check_error(interp, 'AttributeError',
+                f'cannot assign to unsafe attribute {attr}')
+    assert not hasattr(obj, attr)
+
+
 @pytest.mark.parametrize("nested", [False, True])
 def test_names(nested):
     """names test"""
