diff --git a/src/parver/_version.py b/src/parver/_version.py
index e24594e..448553e 100644
--- a/src/parver/_version.py
+++ b/src/parver/_version.py
@@ -164,9 +164,12 @@ class Version:
     :param post_sep2: Specify an alternate separator between the identifier and
         number. The normal form is ``'.'``.
 
-    :param dev_sep: Specify an alternate separator before the development
+    :param dev_sep1: Specify an alternate separator before the development
         release segment. The normal form is ``'.'``.
 
+    :param dev_sep2: Specify an alternate separator between the identifier and
+        number. The normal form is `None`.
+
     :param post_tag: Specify alternate post release identifier `rev` or `r`.
         May be `None` to signify an `implicit post release`_.
 
@@ -254,10 +257,14 @@ class Version:
 
         The seperator between the post release identifier and number.
 
-    .. attribute:: dev_sep
+    .. attribute:: dev_sep1
 
         The separator before the develepment release identifier.
 
+    .. attribute:: dev_sep2
+
+        The seperator between the development release identifier and number.
+
     .. attribute:: post_tag
 
         If this :class:`Version` instance represents a post release, this
@@ -319,7 +326,10 @@ class Version:
     post_sep2: Optional[Separator] = attr.ib(
         default=UNSET, validator=validate_sep_or_unset
     )
-    dev_sep: Optional[Separator] = attr.ib(
+    dev_sep1: Optional[Separator] = attr.ib(
+        default=UNSET, validator=validate_sep_or_unset
+    )
+    dev_sep2: Optional[Separator] = attr.ib(
         default=UNSET, validator=validate_sep_or_unset
     )
     post_tag: Optional[PostTag] = attr.ib(default=UNSET, validator=validate_post_tag)
@@ -423,15 +433,21 @@ class Version:
         assert self.post_sep2 is not UNSET
 
     def _validate_dev(self, set_: Callable[[str, Any], None]) -> None:
+        got_dev_sep1 = self.dev_sep1 is not UNSET
+        got_dev_sep2 = self.dev_sep2 is not UNSET
+
         if self.dev == IMPLICIT_ZERO:
             set_("dev_implicit", True)
             set_("dev", 0)
         elif self.dev is None:
-            if self.dev_sep is not UNSET:
-                raise ValueError("Cannot set dev_sep without dev.")
+            if got_dev_sep1 or got_dev_sep2:
+                raise ValueError("Cannot set dev_sep1 or dev_sep2 without dev.")
+
+        if not got_dev_sep1:
+            set_("dev_sep1", None if self.dev is None else ".")
 
-        if self.dev_sep is UNSET:
-            set_("dev_sep", None if self.dev is None else ".")
+        if not got_dev_sep2:
+            set_("dev_sep2", None)
 
     @classmethod
     def parse(cls, version: str, strict: bool = False) -> "Version":
@@ -478,7 +494,7 @@ class Version:
                 kwargs["post_sep2"] = s.sep2
             elif isinstance(s, segment.Dev):
                 kwargs["dev"] = s.value
-                kwargs["dev_sep"] = s.sep
+                kwargs["dev_sep1"] = s.sep
             elif isinstance(s, segment.Local):
                 kwargs["local"] = s.value
             elif isinstance(s, segment.V):
@@ -531,9 +547,11 @@ class Version:
                 parts.append(str(self.post))
 
         if self.dev is not None:
-            if self.dev_sep is not None:
-                parts.append(self.dev_sep)
+            if self.dev_sep1:
+                parts.append(self.dev_sep1)
             parts.append("dev")
+            if self.dev_sep2:
+                parts.append(self.dev_sep2)
             if not self.dev_implicit:
                 parts.append(str(self.dev))
 
@@ -662,7 +680,8 @@ class Version:
 
         if self.dev is None:
             del d["dev"]
-            del d["dev_sep"]
+            del d["dev_sep1"]
+            del d["dev_sep2"]
 
         return d
 
@@ -680,7 +699,8 @@ class Version:
         pre_sep2: Union[Separator, None, UnsetType] = UNSET,
         post_sep1: Union[Separator, None, UnsetType] = UNSET,
         post_sep2: Union[Separator, None, UnsetType] = UNSET,
-        dev_sep: Union[Separator, None, UnsetType] = UNSET,
+        dev_sep1: Union[Separator, None, UnsetType] = UNSET,
+        dev_sep2: Union[Separator, None, UnsetType] = UNSET,
         post_tag: Union[PostTag, None, UnsetType] = UNSET,
     ) -> "Version":
         """Return a new :class:`Version` instance with the same attributes,
@@ -701,7 +721,8 @@ class Version:
             pre_sep2=pre_sep2,
             post_sep1=post_sep1,
             post_sep2=post_sep2,
-            dev_sep=dev_sep,
+            dev_sep1=dev_sep1,
+            dev_sep2=dev_sep2,
             post_tag=post_tag,
         )
         kwargs = {k: v for k, v in kwargs.items() if v is not UNSET}
@@ -725,7 +746,8 @@ class Version:
             d.pop("pre_sep2", None)
 
         if kwargs.get("dev", UNSET) is None:
-            d.pop("dev_sep", None)
+            d.pop("dev_sep1", None)
+            d.pop("dev_sep2", None)
 
         d.update(kwargs)
         return Version(**d)
