diff --git a/t/unit/test_channel.py b/t/unit/test_channel.py
index 79eb951c7f85351f7ccfff4f1f1089fd88880a22..9a73d167c7f11ab32d8e10b57260a48c5bb866e2 100644
--- a/t/unit/test_channel.py
+++ b/t/unit/test_channel.py
@@ -90,12 +90,22 @@ class test_Channel:
         assert self.c.is_closing is False
         assert self.c.connection is None
 
-    def test_on_close(self):
+    @pytest.mark.parametrize(
+        "channel_is_closing, connection_is_closing, expect_error",
+        [(False, False, True), (True, False, False), (False, True, False)],
+    )
+    def test_on_close(self, channel_is_closing, connection_is_closing, expect_error):
         self.c._do_revive = Mock(name='_do_revive')
-        with pytest.raises(NotFound):
+        self.c.is_closing = channel_is_closing
+        self.conn.is_closing = connection_is_closing
+        if expect_error:
+            with pytest.raises(NotFound):
+                self.c._on_close(404, 'text', 50, 61)
+            self.c._do_revive.assert_called_with()
+        else:
             self.c._on_close(404, 'text', 50, 61)
+            self.c._do_revive.assert_not_called()
         self.c.send_method.assert_called_with(spec.Channel.CloseOk)
-        self.c._do_revive.assert_called_with()
 
     def test_on_close_ok(self):
         self.c.collect = Mock(name='collect')
