diff --git a/spellchecker/spellchecker.py b/spellchecker/spellchecker.py
index 8b4fad4..4f410e8 100644
--- a/spellchecker/spellchecker.py
+++ b/spellchecker/spellchecker.py
@@ -114,7 +114,7 @@ class SpellChecker:
     @distance.setter
     def distance(self, val: int) -> None:
         """set the distance parameter"""
-        tmp = 2
+        tmp = self._distance
         try:
             if 0 < int(val) <= 2:
                 tmp = val
