diff --git a/__tests__/src.typer.ts b/__tests__/src.typer.ts
index a3417ac..82dc44f 100644
--- a/__tests__/src.typer.ts
+++ b/__tests__/src.typer.ts
@@ -92,6 +92,14 @@ describe('typing', () => {
   it('types number with range', () => {
     expect(typer(parse('<number [1,1000]>'))).toMatchObject([{ type: Type.Number }]);
   });
+
+  it('types comma components', () => {
+    expect(typer(parse('something , another-thing'))).toMatchObject([
+      { type: Type.StringLiteral },
+      { type: Type.String },
+      { type: Type.StringLiteral },
+    ]);
+  });
 });
 
 jest.mock('chalk', () => ({}) as typeof import('chalk'));
