diff --git a/src/mdformat_tables/plugin.py b/src/mdformat_tables/plugin.py
index 9636a6e..2f64c66 100644
--- a/src/mdformat_tables/plugin.py
+++ b/src/mdformat_tables/plugin.py
@@ -13,6 +13,7 @@ def add_cli_argument_group(group: argparse._ArgumentGroup) -> None:
     group.add_argument(
         "--compact-tables",
         action="store_true",
+        default=None,
         help="do not add padding to table cells",
     )
 
