diff --git a/typepy/converter/_datetime.py b/typepy/converter/_datetime.py
index f45874c..20d455d 100644
--- a/typepy/converter/_datetime.py
+++ b/typepy/converter/_datetime.py
@@ -3,6 +3,7 @@
 """
 
 from datetime import date, datetime
+from zoneinfo import ZoneInfo
 
 from .._common import strip_ansi_escape
 from .._const import DefaultValue, ParamKey
@@ -56,7 +57,7 @@ class DateTimeConverter(AbstractValueConverter):
 
         if self.__timezone:
             if self.__datetime.tzinfo is None:
-                self.__datetime = self.__timezone.localize(self.__datetime)
+                self.__datetime = self.__datetime.replace(tzinfo=self.__timezone)
             else:
                 self.__datetime = datetime.fromtimestamp(
                     self.__datetime.timestamp(), tz=self.__timezone
@@ -96,7 +97,6 @@ class DateTimeConverter(AbstractValueConverter):
 
     def __from_datetime_string(self):
         import dateutil.parser
-        import pytz
 
         self.__validate_datetime_string()
 
@@ -113,17 +113,16 @@ class DateTimeConverter(AbstractValueConverter):
             raise TypeConversionError(f"failed to parse as a datetime: type={type(self._value)}")
 
         if self.__timezone:
-            pytz_timezone = self.__timezone
+            tzinfo = self.__timezone
         else:
             try:
                 dst_timezone_name = self.__get_dst_timezone_name(self.__get_timedelta_sec())
             except (AttributeError, KeyError):
                 return self.__datetime
 
-            pytz_timezone = pytz.timezone(dst_timezone_name)
+            tzinfo = ZoneInfo(dst_timezone_name)
 
-        self.__datetime = self.__datetime.replace(tzinfo=None)
-        self.__datetime = pytz_timezone.localize(self.__datetime)
+        self.__datetime = self.__datetime.replace(tzinfo=tzinfo)
 
         return self.__datetime
 
