diff --git a/src/smbprotocol/open.py b/src/smbprotocol/open.py
index 2c41de0..4f02628 100644
--- a/src/smbprotocol/open.py
+++ b/src/smbprotocol/open.py
@@ -6,7 +6,7 @@ from collections import OrderedDict
 
 from smbprotocol import MAX_PAYLOAD_SIZE, Dialects
 from smbprotocol.create_contexts import SMB2CreateContextRequest
-from smbprotocol.exceptions import FileClosed, SMBException, SMBUnsupportedFeature
+from smbprotocol.exceptions import FileClosed, SMBException, SMBResponseException, SMBUnsupportedFeature
 from smbprotocol.file_info import (  # While this shouldn't ever be removed, we need to keep this imported so we stay backwards compat. These were; originally defined here.
     FileAttributes,
     FileBothDirectoryInformation,
@@ -1599,6 +1599,10 @@ class Open:
             self._connected = False
             self.tree_connect.session.open_table.pop(self.file_id, None)
             return
+        except SMBResponseException:
+            self._connected = False
+            self.tree_connect.session.open_table.pop(self.file_id, None)
+            raise
 
         c_resp = SMB2CloseResponse()
         c_resp.unpack(response["data"].get_value())
