diff --git a/src/sklearn_compat/_sklearn_compat.py b/src/sklearn_compat/_sklearn_compat.py
index 75d8953..513b45b 100644
--- a/src/sklearn_compat/_sklearn_compat.py
+++ b/src/sklearn_compat/_sklearn_compat.py
@@ -445,6 +445,61 @@ if sklearn_version < parse_version("1.6"):
             **kwargs,
         )
 
+    def check_X_y(
+        X,
+        y,
+        accept_sparse=False,
+        *,
+        accept_large_sparse=True,
+        dtype="numeric",
+        order=None,
+        copy=False,
+        force_writeable=False,
+        ensure_all_finite=None,
+        ensure_2d=True,
+        allow_nd=False,
+        multi_output=False,
+        ensure_min_samples=1,
+        ensure_min_features=1,
+        y_numeric=False,
+        estimator=None,
+    ):
+        """Input validation for standard estimators.
+
+        Check the original documentation for more details:
+        https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_X_y.html
+        """
+        from sklearn.utils.validation import check_X_y as _check_X_y
+
+        if ensure_all_finite is not None:
+            force_all_finite = ensure_all_finite
+        else:
+            force_all_finite = True
+
+        check_X_y_params = inspect.signature(_check_X_y).parameters
+        kwargs = {}
+        if "force_writeable" in check_X_y_params:
+            kwargs["force_writeable"] = force_writeable
+
+        return _check_X_y(
+            X,
+            y,
+            accept_sparse=accept_sparse,
+            accept_large_sparse=accept_large_sparse,
+            dtype=dtype,
+            order=order,
+            copy=copy,
+            force_all_finite=force_all_finite,
+            ensure_2d=ensure_2d,
+            allow_nd=allow_nd,
+            multi_output=multi_output,
+            ensure_min_samples=ensure_min_samples,
+            ensure_min_features=ensure_min_features,
+            y_numeric=y_numeric,
+            estimator=estimator,
+            **kwargs,
+        )
+
     # tags infrastructure
     @dataclass(**_dataclass_args())
     class InputTags:
@@ -749,5 +804,6 @@ else:
         _check_feature_names,  # noqa: F401
         _check_n_features,  # noqa: F401
         check_array,  # noqa: F401
+        check_X_y as check_X_y,
         validate_data,  # noqa: F401
     )
diff --git a/src/sklearn_compat/utils/validation.py b/src/sklearn_compat/utils/validation.py
index 076b400..7baac37 100644
--- a/src/sklearn_compat/utils/validation.py
+++ b/src/sklearn_compat/utils/validation.py
@@ -4,5 +4,6 @@ from sklearn_compat._sklearn_compat import (
     _is_fitted,  # noqa: F401
     _to_object_array,  # noqa: F401
     check_array,  # noqa: F401
+    check_X_y as check_X_y,
     validate_data,  # noqa: F401
 )
