diff --git a/test/objectifier.test.js b/test/objectifier.test.js
index c958fa081368fa47bad36634160cfefeff9c7a47..10216c75ca8d1ccc362bc24eb4872b9b7e6c7ef6 100644
--- a/test/objectifier.test.js
+++ b/test/objectifier.test.js
@@ -147,4 +147,42 @@ test('converts unitless value to number instead of string', () => {
   })
 })
 
+
+test('merges rules ignoring important', () => {
+  let root = parse('a { height: 1px !important; };a { height: 2px }')
+  equal(postcssJS.objectify(root), {
+    a: {
+      height: "2px"
+    }
+  })
+})
+
+test('keeps last important in merge', () => {
+  let root = parse('a { height: 1px !important; };a { height: 2px !important; }')
+  equal(postcssJS.objectify(root), {
+    a: {
+      height: "2px !important"
+    }
+  })
+})
+
+test('prioritizes important in merge', () => {
+  let root = parse('a { height: 1px !important; };a { height: 2px }')
+  equal(postcssJS.objectify(root, true), {
+    a: {
+      height: "1px !important"
+    }
+  })
+})
+
+test('keeps last important with priority', () => {
+  let root = parse('a { height: 1px !important; };a { height: 2px !important; }')
+  equal(postcssJS.objectify(root, true), {
+    a: {
+      height: "2px !important"
+    }
+  })
+})
+
+
 test.run()
