diff --git a/tests/unit/test_checker_manager.py b/tests/unit/test_checker_manager.py
index 68dd82aa226077bd0453023458c0c9cd40f5f1f7..8d6b3dcfea00a561917eb9bb5798d4591a30a424 100644
--- a/tests/unit/test_checker_manager.py
+++ b/tests/unit/test_checker_manager.py
@@ -61,6 +61,16 @@ def test_multiprocessing_cpu_count_not_implemented():
     assert manager.jobs == 0
 
 
+def test_jobs_count_limited_to_file_count():
+    style_guide = style_guide_mock()
+    style_guide.options.jobs = JobsArgument("4")
+    style_guide.options.filenames = ["file1", "file2"]
+    manager = checker.Manager(style_guide, finder.Checkers([], [], []), [])
+    assert manager.jobs == 4
+    manager.start()
+    assert manager.jobs == 2
+
+
 def test_make_checkers():
     """Verify that we create a list of FileChecker instances."""
     style_guide = style_guide_mock()
